toolchain/sem_ir/inst.h

```diff
@@ -92,12 +140,17 @@ struct InstLikeTypeInfo<InstCat> : InstLikeTypeInfoBase<InstCat> {
     RawStringOstream out;
     out << "{";
     llvm::ListSeparator sep;
-    for (auto kind : InstCat::Kinds) {
+    for (auto kind : InstCat::CategoryInfo::Kinds) {
       out << sep << kind;
     }
     out << "}";
     return out.TakeStr();
   }
+
+ private:
+  // Trigger validation of `InstCat`.
+  static constexpr auto IgnoreThis =
+      ValidateCategory<InstCat>(typename InstCat::CategoryInfo());
```

Comment:
Optional: rather than a static data member, I'd be tempted to use a `static_assert` instead here (and change `ValidateCategory` to return a `bool` that's always `true`):
```suggestion
  // Trigger validation of `InstCat`.
  static_assert(ValidateCategory<InstCat>(typename InstCat::CategoryInfo()));
```
I think that carries the implication slightly better that we're asserting a property of the `InstCat` (even though *this* `static_assert` isn't actually the one doing the asserting).

Reply:
Yeah, that's probably the lesser evil here.